internal/client/cloudfoundry/client.go

```diff
@@ -27,5 +117,495 @@ func NewClient(config *config.Config, logger *gosteno.Logger) (*cfclient.Client,
+	for _, space := range spaces {
+		spacesPerGuid[space.Guid] = space
+	}
+
+	// Fetch orgs
+	orgsPerGuid := map[string]cfclient.Org{}
+	q := url.Values{}
+	q.Set("results-per-page", "100") // 100 is the max
+	orgs, err := cfc.client.ListOrgsByQuery(q)
+	if err != nil {
+		cfc.logger.Error(err.Error())
+		return nil, err
+	}
+	// Create an org Map
+	for _, org := range orgs {
+		orgsPerGuid[org.Guid] = org
+	}
+
+	// Populate CFApplication
+	results := []CFApplication{}
+	for _, cfapp := range cfapps {
+		updatedApp := cfapp
+		appGUID := cfapp.GUID
+		spaceGUID := cfapp.SpaceGUID
+		processes, exists := processesPerApp[appGUID]
+		if exists {
+			updatedApp.setV3ProcessData(processes)
+		}
+		space, exists := spacesPerGuid[spaceGUID]
+		if exists {
+			updatedApp.setV3SpaceData(space)
+		}
+		orgGUID := updatedApp.OrgGUID
+		org, exists := orgsPerGuid[orgGUID]
+		if exists {
+			updatedApp.setV3OrgData(org)
+		}
+		results = append(results, updatedApp)
+	}
+
+	return results, nil
+}
+
+
+func (cfc *CFClient) getV3ApplicationsByPage(page int) ([]CFApplication, int, error){
+	q := url.Values{}
+	q.Set("per-page", "5000") // 5000 is the max
+
+	// Since we don't know what version of the v3 endpoint we use, we have to discover it.
+	// Hence, we need to try for the latest to the oldest approach.
+	// 1. try with include=space,space.organization (as supported in 3.77.0)
+	// 2. try with include=space,org as supported in 3.76.0 all the way down to 3.73.0
+	// 3. try with include=space only since it seems that org doesn't work on some versions
+	//	- In this case we need to make a separate call to the org endpoint (ideally fetch quotas at the same time).
+	// 4. try without include
+	//  - In this case we need to make two more calls, on for org and one for space.
+	//
+	//if cfc.include != "" {
+	//	q.Set("include", cfc.include)
+	//}
+
+	r := cfc.client.NewRequest("GET", "/v3/apps?" + q.Encode())
+	resp, err := cfc.client.DoRequest(r)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error requesting apps page %d", err)
+	}
+	// Read body response
+	defer resp.Body.Close()
+	resBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error reading app response for page %d", page)
+	}
+	// Unmarshal body response into v3AppResponse objects
+	var appResp v3AppResponse
+	err = json.Unmarshal(resBody, &appResp)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error unmarshalling app response for page %d", page)
+	}
+	// Create CFApplication objects
+	appResources := appResp.Resources
+	results := []CFApplication{}
+	for _, app := range appResources {
+		cfapp := CFApplication{}
+		cfapp.setV3AppData(app)
+		results = append(results, cfapp)
+	}
+
+	return results, appResp.Pagination.TotalPages, nil
+}
+
+func (cfc *CFClient) getV3Processes() ([]cfclient.Process, error){
+	q := url.Values{}
+	q.Set("per_page", "5000") // 5000 is the max
+	return cfc.client.ListAllProcessesByQuery(q)
+}
```

Comment:
Why don't we have the same logic as the other endpoints to parallelize page fetching ?